src/peft/tuners/tuners_utils.py

```diff
@@ -32,6 +35,47 @@
 logger = logging.getLogger(__name__)
 
 
+@contextmanager
+def onload_layer(layer):
+    for name, module in layer.named_modules():
```

Comment:
Since this is the same loop as before, how about storing these modules in a variable and re-using it below? So something along the lines of:

```python
modules_with_align_device_hooks = [m for name, m in layer.named_modules() if (name not in ["", "base_layer"]) and hasattr(module, "_hf_hook") and isinstance(module._hf_hook, AlignDevicesHook) and module._hf_hook.offload]
```

The long conditional could be factored out into a local function or something like that for readability.